pulp_2_tests/tests/rpm/api_v2/test_modularity.py

```diff
@@ -430,3 +431,71 @@ def test_all(self):
             RPM_UNSIGNED_FEED_COUNT - sum(MODULE_FIXTURES_PACKAGES.values()),
             non_modular_units
         )
+
+
+class CheckModulesYamlTestCase(unittest.TestCase):
+    """Check whether modules.yaml is available if appropriate."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class wide variables."""
+        cls.cfg = config.get_config()
+        if cls.cfg.pulp_version < Version('2.18'):
+            raise unittest.SkipTest('This test requires Pulp 2.18 or newer.')
+        cls.client = api.Client(cls.cfg, api.json_handler)
+
+    def test_no_modules_yaml_generated_non_modular(self):
+        """Verify no modules yaml is generated for non modular content.
+
+        This test does the following
+
+        1. Create and sync a non-modular content.
+        2. Publish the synced content.
+        3. Check whether no modules.yaml is generated for published content.
+
+        This test case targets:
+
+        * `Pulp #4350 <https://pulp.plan.io/issues/4350>`_.
```

Comment:
Add the issue itself.
* `Pulp #4352 <https://pulp.plan.io/issues/4352>`_.

Reply:
no @ragabala, it was a typo. It was supposed to be ``4252``.

Reply:
Have added the change !